src/include/server/mir/input/input_event_transformer.h

```diff
@@ -52,15 +52,33 @@ class InputEventTransformer : public EventFilter
         virtual bool transform_input_event(EventDispatcher const& dispatcher, EventBuilder*,  MirEvent const&) = 0;
     };
 
+    struct Registration
+    {
+        Registration(
+            InputEventTransformer* event_transformer,
+            std::shared_ptr<mir::input::InputEventTransformer::Transformer> transformer);
+
+        ~Registration();
+
+        Registration(Registration&& other) = default;
```

Comment:
I don't think the `std::function` move constructor does what we need here.

Specifically, we need the "source" `unregister` to hold a no-op. But I don't think that is what the library specification gives us. By my reading it gives a "valid but unspecified state".